src/player.py

```diff
@@ -139,12 +50,13 @@ def respawn(self):
         chosen_spawn_point = self.initial_spawn_points[
             random.randrange(len(self.initial_spawn_points))
         ].transform
-        set_sprite_location(self.sprite, chosen_spawn_point)
+        self.vehicle.location = chosen_spawn_point
+        self.vehicle.movement.reset_velocity()
 
     @property
-    def location(self):
-        return get_sprite_location(self.sprite)
+    def input(self):
+        return self.vehicle.input
```

Comment:
Is `self.vehicle.input` the source of truth here?  And `player` grabs its own `input` from there?  Naively I would assume should be the other way around.

Reply:
Fixed. Got rid of the property. Now player has an input and vehicle calls self.player.input if it need it